SilKit/source/services/orchestration/TimeSyncService.cpp

```diff
@@ -499,13 +499,26 @@ void TimeSyncService::ExecuteSimStep(std::chrono::nanoseconds timePoint, std::ch
     _watchDog.Start();
     _simTask(timePoint, duration);
     _watchDog.Reset();
+
+    _simStepCounterMetric->Add(1);
+
+    if (!IsBlocking())
```

Comment:
Isn't the `if` condition inverted? I.e., the step immediately completes if the step is sync, since `IsBlocking() == IsSimStepSync()`.

Reply:
Yes got this one wrong